src/wallet/spend.cpp

```diff
@@ -1188,47 +1188,25 @@ bool FundTransaction(CWallet& wallet, CMutableTransaction& tx, CAmount& nFeeRet,
                 // The input was not in the wallet, but is in the UTXO set, so select as external
                 preset_txin.SetTxOut(coins[outPoint].out);
             } else {
-                error = _("Unable to find UTXO for external input");
-                return false;
+                return util::Error{_("Unable to find UTXO for external input")};
             }
         }
         preset_txin.SetSequence(txin.nSequence);
         preset_txin.SetScriptSig(txin.scriptSig);
         preset_txin.SetScriptWitness(txin.scriptWitness);
     }
 
-    auto res = CreateTransaction(wallet, vecSend, nChangePosInOut == -1 ? std::nullopt : std::optional<unsigned int>((unsigned int)nChangePosInOut), coinControl, false);
+    auto res = CreateTransaction(wallet, vecSend, change_pos, coinControl, false);
     if (!res) {
-        error = util::ErrorString(res);
-        return false;
-    }
-    const auto& txr = *res;
-    CTransactionRef tx_new = txr.tx;
-    nFeeRet = txr.fee;
-    nChangePosInOut = txr.change_pos ? *txr.change_pos : -1;
-
-    if (nChangePosInOut != -1) {
-        tx.vout.insert(tx.vout.begin() + nChangePosInOut, tx_new->vout[nChangePosInOut]);
-    }
-
-    // Copy output sizes from new transaction; they may have had the fee
-    // subtracted from them.
-    for (unsigned int idx = 0; idx < tx.vout.size(); idx++) {
-        tx.vout[idx].nValue = tx_new->vout[idx].nValue;
```

Comment:
I think you misunderstand what the original code was doing. It isn't preserving the original values, it's replacing the original values with the values that `CreateTransaction` produces. Originally `FundTransaction` is calling `CreateTranasction` and copying specific parts of the result into the original transaction and returning that modified original. The change I've made here is to just return the result of `CreateTransaction` directly, which matches the previous behavior because the earlier commits are preserving all of the components of the transaction that `FundTranasction` was preserving.

Reply:
oops, yeah..  nvm then.